L74-java-channel-creds.md

```diff
@@ -0,0 +1,489 @@
+Java: Channel and Server Credentials
+----
+* Author(s): [Eric Anderson](https://github.com/ejona86)
+* Approver: sanjaypujare
+* Status: In Review {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2020-06-17
+* Discussion at: https://groups.google.com/g/grpc-io/c/GNU9svnDsQs/m/kG6XsUy9BwAJ
+
+## Abstract
+
+Develop an API that separates transport credential construction from
+channel/server construction.
+
+This is to allow:
+
+*   implementing new transport credentials without creating a new
+    channel/server type, which allows users of the API to separate concerns
+*   providing stable APIs for advanced credential configuration like mutual
+    TLS, without polluting the general-purpose ManagedChannelBuilder API with
+    many overloads that don't apply to all transports
+*   composition of credentials, like necessary for xDS where a fallback
```

Comment:
Even without xDS the `CompositeChannelCredentials`  feature also provides composition of credentials which is quite useful.

Reply:
CompositeChannelCredentials does not compose ChannelCredentials. But that's the opposite of what this is talking about. This is listing requirements/goals, and xDS benefits from composition. CompositeChannelCredentials is a _solution_, so by definition it isn't useful to describe requirements.